x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/NativeUsersStore.java

```diff
@@ -283,6 +284,34 @@ private void createReservedUser(String username, char[] passwordHash, RefreshPol
         });
     }
 
+    /**
+     * Asynchronous method to create a reserved user with the given password hash. This should fail when the document for the
+     * specified user already exists in the security index
+     */
+    public void createReservedUser(String username, char[] passwordHash, ActionListener<Void> listener) {
+        securityIndex.prepareIndexIfNeededThenExecute(listener::onFailure, () -> {
+            executeAsyncWithOrigin(
+                client.threadPool().getThreadContext(),
+                SECURITY_ORIGIN,
+                client.prepareIndex(SECURITY_MAIN_ALIAS)
+                    .setOpType(DocWriteRequest.OpType.CREATE)
+                    .setId(getIdForUser(RESERVED_USER_TYPE, username))
+                    .setSource(
+                        Fields.PASSWORD.getPreferredName(),
+                        String.valueOf(passwordHash),
+                        Fields.ENABLED.getPreferredName(),
+                        true,
+                        Fields.TYPE.getPreferredName(),
+                        RESERVED_USER_TYPE
+                    )
+                    .setRefreshPolicy(RefreshPolicy.IMMEDIATE)
+                    .request(),
+                listener.<IndexResponse>delegateFailure((l, indexResponse) -> l.onResponse(null)),
+                client::index
+            );
+        });
+    }
```

Comment:
Is this 90% identical to the `updateReservedUser` method?

The main difference is just the `opType`, and it doesn't feel like we need to duplicate all the code for that.

Reply:
Yes. Late night interface design is the worst. I'll add the opType as a parameter since also https://github.com/elastic/elasticsearch/pull/77291/files#r702622693 makes sense

Reply:
79a6c21